custom_components/tahoma/lock.py

```diff
@@ -46,49 +43,44 @@ def update(self):
             return
 
         self.controller.get_states([self.tahoma_device])
-        self._battery_level = self.tahoma_device.active_states["core:BatteryState"]
-        self._name = self.tahoma_device.active_states["core:NameState"]
-        if (
-            self.tahoma_device.active_states.get("core:LockedUnlockedState")
-            == STATE_LOCKED
-        ):
-            self._lock_status = STATE_LOCKED
-        else:
-            self._lock_status = STATE_UNLOCKED
-        self._available = (
-            self.tahoma_device.active_states.get("core:AvailabilityState")
-            == "available"
-        )
 
     def unlock(self, **kwargs):
         """Unlock method."""
-        _LOGGER.debug("Unlocking %s", self._name)
+        _LOGGER.debug("Unlocking %s", self.name)
         self.apply_action("unlock")
 
     def lock(self, **kwargs):
         """Lock method."""
-        _LOGGER.debug("Locking %s", self._name)
+        _LOGGER.debug("Locking %s", self.name)
         self.apply_action("lock")
 
     @property
     def name(self):
         """Return the name of the lock."""
-        return self._name
+        return self.tahoma_device.active_states["core:NameState"]
 
     @property
     def available(self):
         """Return True if the lock is available."""
-        return self._available
+        return (
+            self.tahoma_device.active_states.get("core:AvailabilityState")
+            == "available"
```

Comment:
`available()` should be inherited from tahoma_device

Reply:
core:AvailabilityState exists for all the devices?

Reply:
I don't think so. That's why I left this one in lock.py

Reply:
I believe it exists for more devices, since it exists in `tahoma_device`. I would need to check.

Reply:
That's the same check that the one in TahomaDevice. I've removed this one.